contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol

```diff
@@ -141,6 +141,58 @@ contract KeeperRegistrar2_0 is TypeAndVersionInterface, ConfirmedOwner, ERC677Re
     uint8 source,
     address sender
   ) external onlyLINK {
+    _register(name, encryptedEmail, upkeepContract, gasLimit, adminAddress, checkData, amount, source, sender);
+  }
+
+  /**
+   * @notice Allows external users to register upkeeps
+   * @notice registerUpkeep can only be called through transferAndCall on LINK contract
+   * @param name string of the upkeep to be registered
+   * @param encryptedEmail email address of upkeep contact
+   * @param upkeepContract address to perform upkeep on
+   * @param gasLimit amount of gas to provide the target contract when performing upkeep
+   * @param adminAddress address to cancel upkeep and withdraw remaining funds
+   * @param checkData data passed to the contract when checking for upkeep
+   * @param amount quantity of LINK upkeep is funded with (specified in Juels)
+   * @param source application sending this request
+   */
+  function registerUpkeep(
+    string memory name,
+    bytes calldata encryptedEmail,
+    address upkeepContract,
+    uint32 gasLimit,
+    address adminAddress,
+    bytes calldata checkData,
+    uint96 amount,
+    uint8 source
+  ) external returns (uint256) {
+    if (amount < s_config.minLINKJuels) {
+      revert InsufficientPayment();
+    }
+
+    KeeperRegistryBaseInterface keeperRegistry = s_config.keeperRegistry;
+
+    bool success = LINK.transferFrom(msg.sender, owner(), amount);
+    if (!success) {
+      revert LinkTransferFailed(address(keeperRegistry));
```

Comment:
this should be `address(this)`.

Reply:
instead of `owner()`? I was questioning that one too.